### DISCUSSION TITLE
Easy external acces

### DISCUSSION THREAD

**Discussion by FreddGit** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#787] [Q&A]
Hello, first of all, soory for my english. I have juste one question about external acces to the wifi module. When we buy a spa with wifi, or something with external acces it seems very easy. We download an app for android on the play store, the app detect our spa and voila. Nothing to do with router, NAT, open port or other. How does it work and why can't we do that with that module? Thank you

**Comment by visualapproach** [comment]
Hi! We can but we don't want to rely on someone else's computer (cloud server) to work and not spy on us.

**Comment by metaljay** [comment]
The easiest option (if you have an iPhone) is to install something like homebridge or home assistant on a raspberry pi, then use MQTT to toggle the hot tub. This way your using Apple as the secure link to your home network.

**Comment by FreddGit** [comment]
Thank you for your answers but my question is more about how this app can communicate with the spa (by example a layz spa with built in wifi)? There is no configuration on the router... for homeassistant or this wifi module we need to open a port, or have a vpn on the router...

**Reply by metaljay** [reply]
And I get that, but I’m saying I don’t need todo anything like that.
Simply connecting home assistant (via MQTT) to Apple HomeKit I can control and manage my hot tub remotely. See my screenshot (on 5G)
![IMG_5091](https://github.com/user-attachments/assets/7c7f039e-7760-4580-b151-327b27141caf)

**Reply by FreddGit** [reply]
ok, but you need homeassistant and homekit...

**Reply by metaljay** [reply]
> ok, but you need homeassistant and homekit...

And? Much easier and safer than opening ports or using someone else’s server like an app would? Just buy a raspberry pi (£30?) and call it a day

**Comment by FreddGit** [comment]
my question is how the others do? how is it possible with a spa with wifi and an app on android? I just want to interract with the wifi module, nothing else, so why a raspbarry pi or a domotic box just to do that? In France we have by example an internet operator like free. On the router it is possible to have a VPN, wireguard and with the app wireguard on android and chrome it is possible to interract with the wifi module but it doesn't anwser to my question.

**Reply by metaljay** [reply]
Have you read the manual?

**Reply by FreddGit** [reply]
yes, i have read the manual, and find no answer to my question. If you, show me.

**Reply by metaljay** [reply]
Check the end sections about homebridge MQTT and home assistant. They are the only supported ways to integrate. No custom app or external server.

**Reply by FreddGit** [reply]
yes, i never tell something else... my question was how do the others... Like a discussion i was thinking it was something interesting... I think that th e code of the module is made to evoluate.

**Comment by visualapproach** [comment]
I think the OP's question is "how do other projects connect to their devices from the internet (without hassle)".
I assume he can reach his spa module from within his own network.

The answer is that the other's device's "apps" on your phone connects to a server somewhere. Most likely in China. The Devices also connects to that server and all communications is made over that server.
Can we do that too? Yes. But who will pay for, and trust that server is online all the time? I won't. It also has a lot of negative sides - it will take more time for the data to reach it's goal. It won't work without internet or even without that server. Security and so on.
So if you want to use this project from the internet *you* have to arrange a solution. @metaljay suggested a few.
